AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java

```diff
@@ -446,7 +448,7 @@ public void onClick(View v) {
         final ArrayList<String> deckNames = new ArrayList<>();
 
         ArrayList<JSONObject> decks = getCol().getDecks().all();
-        Collections.sort(decks, new JSONNameComparator());
+        Collections.sort(decks, DeckComparator.instance);
```

Comment:
This is replaced by the wrong comparator I think, this should be 
```suggestion
        Collections.sort(decks, DeckNameComparator.instance);
```

Reply:
If it were the case, the build would fail with a type error.
However, i may be more than happy if you can suggest better names.
DeckNameComparator is a comparator for strings, assuming that those strings are deck's names.
DeckComparator are comparator for decks according to their name.
Both are required, because you have some instances of list of names, and some instances of list of decks.

Reply:
By my read, you extracted comparators (good), and then  you called the extracted comparators, just that for this one you chose the wrong extracted comparator. I just re-checked it. I still believe my read is correct. The one you are calling now is a different implementation than was used before, the implementation used before was extracted to the one I suggest.

Reply:
I did a rebase on master and didn't change anything else, so I am surprised that you find something so important changed.
It is however possible that I did already corrected it on my computer and just forgot to push it. I'd be surprised, because emacs warns me when I've got commit not pushed, but that's not impossible